### ISSUE TITLE
Message draft reappearing 

### ISSUE THREAD

**Issue by riyapenn** [WhisperSystems/Signal-iOS#1537]
- [X] I have searched open and closed issues for duplicates

----------------------------------------

### Bug description
Message draft reappears in message input field even if message was sent in the past

### Steps to reproduce -- intermittent, not consistent
- Compose message
- send message
- leave/close out of Signal at some point
- open Signal and view a conversation thread
- see draft of a message

**Actual result:** See draft of a message, this happens for some threads and is not consistent
**Expected result:** No draft appears if message was sent

### Device info

**Device**: iPhone 6S, 6
**iOS version**: 10.0.2, 10.1.1
**Signal version:** 2.6.7

### Link to debug log
User 1: https://gist.github.com/anonymous/c41f100e7868310d917cc2edf13b8f10
User 2: https://gist.github.com/anonymous/2e6a42d1aa29c2f10e224aeeb874ddb2

**User 1 potentially relevant information here:** 
**Original message written:**
```
2016/12/07 22:13:39:295  Error in websocket heartbeat: Invalid State: Cannot call `sendPing:error:` until connection is open.
2016/12/07 22:14:04:670  [TSAttachmentStream] Created file at /var/mobile/Containers/Data/Application/1150D4C0-75C1-4F49-A23A-5892CF925BF7/Documents/Attachments/6BC5D6BA-D26D-4D07-B98D-E80CFD8DD555.jpeg
2016/12/07 22:14:04:699  [TSNetworkManager] Making request: <TSAllocAttachmentRequest: 0x170637ac0> { URL: v1/attachments }
2016/12/07 22:14:06:695  [OWSUploadingService] Uploaded attachment.
2016/12/07 22:14:06:774  [TSNetworkManager] Making request: <TSSubmitMessageRequest: 0x17443f740> { URL: v1/messages/[ REDACTED_PHONE_NUMBER ] }
2016/12/07 22:14:07:164  [TSNetworkManager] Making request: <TSSubmitMessageRequest: 0x17443eb00> { URL: v1/messages/[ REDACTED_PHONE_NUMBER ] }
2016/12/07 22:14:07:289  Succesfully sent sync transcript.
2016/12/07 22:14:09:833  [TSNetworkManager] Making request: <TSSubmitMessageRequest: 0x1708250a0> { URL: v1/messages/[ REDACTED_PHONE_NUMBER ] }
2016/12/07 22:14:09:947  [TSNetworkManager] Making request: <TSSubmitMessageRequest: 0x1708257e0> { URL: v1/messages/[ REDACTED_PHONE_NUMBER ] }
2016/12/07 22:14:09:948  [MessagesViewController] Successfully sent message.
2016/12/07 22:14:10:099  Succesfully sent sync transcript.
2016/12/07 22:14:11:302  Got message with verb: PUT and path: /api/v1/message
2016/12/07 22:14:11:303  Received a delivery receipt
2016/12/07 22:14:11:410  Got message with verb: PUT and path: /api/v1/message
2016/12/07 22:14:11:410  Received a delivery receipt
2016/12/07 22:16:18:555  Error in websocket heartbeat: Invalid State: Cannot call `sendPing:error:` until connection is open.
2016/12/07 22:16:18:909  Got message with verb: PUT and path: /api/v1/message
2016/12/07 22:16:47:360  Error in websocket heartbeat: Invalid State: Cannot call `sendPing:error:` until connection is open.

```
**Draft re-appearing:**
Sometime toward the end of log of User 1.

**User 2 potentially relevant information here:**

**Original message written:**
```
2016/12/04 21:50:50:151  [VersionMigrations] Checking migrations. currentVersion: 2.6.6.7, lastRanVersion: 2.6.6.7
2016/12/04 21:50:50:534  The app was launched from inactive
2016/12/04 21:50:50:676  [TSNetworkManager] Making request: <TSAvailablePreKeysCountRequest: 0x17402b020> { URL: v2/keys }
2016/12/04 21:50:51:045  [TSNetworkManager] Making request: <TSContactsIntersectionRequest: 0x174427fa0> { URL: v1/directory/tokens }
2016/12/04 21:50:51:051  [TSNetworkManager] Making request: <TSContactsIntersectionRequest: 0x174431200> { URL: v1/directory/tokens }
2016/12/04 21:50:51:351  [ContactsUpdater] successfully intersected contacts.
2016/12/04 21:50:51:351  [OWSContactsManager] Successfully intersected contacts.
2016/12/04 21:50:51:360  [ContactsUpdater] successfully intersected contacts.
2016/12/04 21:50:51:360  [OWSContactsManager] Successfully intersected contacts.
2016/12/04 21:51:19:714  [TSAttachmentStream] Created file at /var/mobile/Containers/Data/Application/24BAA65D-03C2-4DB2-A49B-4C2F89CBADB8/Documents/Attachments/C47E428D-1201-40F5-8B42-CBF8EDDF6EC8.jpeg
2016/12/04 21:51:19:787  [TSNetworkManager] Making request: <TSAllocAttachmentRequest: 0x17003c5e0> { URL: v1/attachments }
2016/12/04 21:51:20:860  [OWSUploadingService] Uploaded attachment.
2016/12/04 21:51:20:965  [TSNetworkManager] Making request: <TSSubmitMessageRequest: 0x17062a7a0> { URL: v1/messages/[ REDACTED_PHONE_NUMBER ] }
2016/12/04 21:51:21:112  [TSNetworkManager] Making request: <TSSubmitMessageRequest: 0x17003c4e0> { URL: v1/messages/[ REDACTED_PHONE_NUMBER ] }
2016/12/04 21:51:21:217  Succesfully sent sync transcript.
2016/12/04 21:51:26:234  [TSNetworkManager] Making request: <TSSubmitMessageRequest: 0x17463c160> { URL: v1/messages/[ REDACTED_PHONE_NUMBER ] }
2016/12/04 21:51:26:509  [TSNetworkManager] Making request: <TSSubmitMessageRequest: 0x174234680> { URL: v1/messages/[ REDACTED_PHONE_NUMBER ] }
2016/12/04 21:51:26:510  [MessagesViewController] Successfully sent message.
2016/12/04 21:51:26:611  Succesfully sent sync transcript.
```

**Draft re-appearing:**
```
2016/12/07 22:31:03:856  [VersionMigrations] Checking migrations. currentVersion: 2.6.7.4, lastRanVersion: 2.6.6.7
2016/12/07 22:31:04:130  The app was launched from inactive
2016/12/07 22:31:04:190  [TSNetworkManager] Making request: <TSAvailablePreKeysCountRequest: 0x174039d40> { URL: v2/keys }
2016/12/07 22:31:04:923  [TSNetworkManager] Making request: <TSContactsIntersectionRequest: 0x17443bfe0> { URL: v1/directory/tokens }
2016/12/07 22:31:04:938  [TSNetworkManager] Making request: <TSContactsIntersectionRequest: 0x17443f080> { URL: v1/directory/tokens }
2016/12/07 22:31:20:334  [ContactsUpdater] successfully intersected contacts.
2016/12/07 22:31:20:334  [OWSContactsManager] Successfully intersected contacts.
2016/12/07 22:31:20:705  [ContactsUpdater] successfully intersected contacts.
2016/12/07 22:31:20:705  [OWSContactsManager] Successfully intersected contacts.
2016/12/07 22:32:09:042  Uploading logs failed with error: Error Domain=NSURLErrorDomain Code=-1001 "The request timed out." UserInfo={NSUnderlyingError=0x170640060 {Error Domain=kCFErrorDomainCFNetwork Code=-1001 "The request timed out." UserInfo={NSErrorFailingURLStringKey=https://api.github.com/gists, NSErrorFailingURLKey=https://api.github.com/gists, _kCFStreamErrorCodeKey=-2102, _kCFStreamErrorDomainKey=4, NSLocalizedDescription=The request timed out.}}, NSErrorFailingURLStringKey=https://api.github.com/gists, NSErrorFailingURLKey=https://api.github.com/gists, _kCFStreamErrorDomainKey=4, _kCFStreamErrorCodeKey=-2102, NSLocalizedDescription=The request timed out.}
```

**Comment by riyapenn** [comment]
https://gist.github.com/3ca90737a4e128ffffa7ae9437896b04 
Timeline: 
Original message draft composed a couple of days ago, never sent
Sent message draft some time today
Draft still reappears

**Comment by riyapenn** [comment]
Log URL: https://gist.github.com/49ec01b61b8a707dd5e9d9e1161bbeb7 
Tell us about the issue:

It happened 16.03. 
I sent 4 more messages after the one reappearing.

I believe I wrote it, pressed power button to put iPhone to sleep before I logged in once again to send it.

**Comment by charlesmchen** [comment]
This is fixed and will ship with v2.12.
